tests/python/pants_test/reporting/test_json_reporter.py

```diff
@@ -0,0 +1,160 @@
+      'workunits': {
+        'root_id': {
+          'name': 'root',
+          'id': 'root_id',
+          'parent_name': '',
+          'parent_id': '',
+          'labels': [
+            'IAMROOT'
+          ],
+          'cmd': '',
+          'start_time': -5419800.0,
+          'outputs': {},
+          'children': [
+            {
+              'name': 'child1',
+              'id': 'child1_id',
+              'parent_name': 'root',
+              'parent_id': 'root_id',
+              'labels': [],
+              'cmd': '',
+              'start_time': 31564800.0,
+              'outputs': {},
+              'children': [
+                {
+                  'name': 'grandchild',
+                  'id': 'grandchild_id',
+                  'parent_name': 'child1',
+                  'parent_id': 'child1_id',
+                  'labels': [
+                    'LABEL1'
+                  ],
+                  'cmd': '',
+                  'start_time': 479721600.0,
+                  'outputs': {},
+                  'children': [],
+                  'log_entries': [],
+                  'outcome': 'SUCCESS',
+                  'end_time': 479721610.0,
+                  'unaccounted_time': 0
+                }
+              ],
+              'log_entries': [],
+              'outcome': 'SUCCESS',
+              'end_time': 31564810.0,
+              'unaccounted_time': 0
+            },
+            {
+              'name': 'child2',
+              'id': 'child2_id',
+              'parent_name': 'root',
+              'parent_id': 'root_id',
+              'labels': [],
+              'cmd': '',
+              'start_time': 684140400.0,
+              'outputs': {},
+              'children': [],
+              'log_entries': [],
+              'outcome': 'SUCCESS',
+              'end_time': 684140410.0,
+              'unaccounted_time': 0
+            }
+          ],
+          'log_entries': [],
+          'outcome': 'SUCCESS',
+          'end_time': -5419790.0,
+          'unaccounted_time': 0
+        },
+      },
+      'artifact_cache_stats': []
+    }
+
+    with temporary_dir() as temp_dir:
+      run_tracker = FakeRunTracker()
+      reporter = JsonReporter(run_tracker, JsonReporter.Settings(log_level=Report.INFO, json_dir=temp_dir))
+      run_tracker.reporter = reporter
+
+      run_tracker.start()
+
+      with run_tracker.new_workunit(FakeWorkUnit(None, 'root', labels=['IAMROOT'], start_time=-5419800.0)) as root_workunit:
+        with run_tracker.new_workunit(FakeWorkUnit(root_workunit, 'child1', start_time=31564800.0)) as child1:
+          with run_tracker.new_workunit(FakeWorkUnit(child1, 'grandchild', labels=['LABEL1'], start_time=479721600.0)): pass
+        with run_tracker.new_workunit(FakeWorkUnit(root_workunit, 'child2', start_time=684140400.0)): pass
+
+      run_tracker.end()
+
+      result = json.loads(open(os.path.join(reporter.report_path())).read())
+      self.assertDictEqual(expected, result)
```

Comment:
I'd like to see some more test cases, maybe with different workunit shapes?

Reply:
Sure thing - I just added another for further nesting, what others do you think would be helpful to cover here?